### DISCUSSION TITLE
OpenID Connect button not shown for login - login not possible

### DISCUSSION THREAD

**Discussion by sseide** [photoprism/photoprism#4435] [Q&A] [answered]
Hello,
i want to test Photoprism and set up a test instance with Docker Compose. System working so far but a cannot get the OpenID Connect integration working...

My OIDC Provider is a Nextcloud instance (its working fine a OIDC provider with other apps). I set all OIDC related env vars as documented in https://docs.photoprism.app/getting-started/advanced/openid-connect/#identity-providers and restarted the system. Checking the running docker config it seems to be configured fine but the OpenID Connect button is not displayed on the login page - therefor i cannot login via OIDC. The only user able to login is the default "admin" user.

What did i wrong?

The login page does not look like the one in the documentation here: 
https://docs.photoprism.app/getting-started/advanced/openid-connect/#identity-providers
Its only displaying the username/password field and the button "Sign In". All other buttons are missing.

Booth Photoprism and Nextcloud have valid TLS certificates and can be reached via HTTPS. Booth use their own domain name. (DNS names changed here)

inside the docker-compose.yml i set the following OIDC related env vars:
```yml
    environment:
     PHOTOPRISM_SITE_URL: "http://photo.nextcloud.my.domain/" 
      PHOTOPRISM_OIDC_CLIENT: "S..."
      PHOTOPRISM_OIDC_SECRET: "l..."
      PHOTOPRISM_OIDC_REDIRECT: "false"
      PHOTOPRISM_OIDC_PROVIDER: "Nextcloud"
      PHOTOPRISM_OIDC_URI: "https://nextcloud.my.domain/"
      PHOTOPRISM_OIDC_USERNAME: "email"
      PHOTOPRISM_OIDC_REGISTER: "true"
```

Querying the active config from the docker instance lists the following OIDC config:
```
$ docker compose exec photoprism photoprism show config

OPENID CONNECT (OIDC)

|---------------|------------------------------------------------------------------|
|     Name      |                              Value                               |
|---------------|------------------------------------------------------------------|
| oidc-uri      | https://nextcloud.my.domain/                                     |
| oidc-client   | S...                                                             |
| oidc-secret   | **************************************************************   |
| oidc-scopes   | openid email profile address                                     |
| oidc-provider | Nextcloud                                                        |
| oidc-icon     |                                                                  |
| oidc-redirect | false                                                            |
| oidc-register | true                                                             |
| oidc-username | email                                                            |
| oidc-role     | guest                                                            |
| oidc-webdav   | false                                                            |
| disable-oidc  | false                                                            |
|---------------|------------------------------------------------------------------|
```

I also tested with "oidc-redirect true" and "oidc-register false" but it does not change anything it seems.

**Big Question - What to do to activate OIDC for user login?**

Thanks,
S. Seide

_and a remark about logging:_

I activated the log level "TRACE" and set DEBUG "true" but it is not writing more log messages at all in the log for this case. The current logging implementation is completly useless to debug such issues, there is no information on which data store it is searching a user, or if something else goes wrong or client is invalid or ...
And (from security point of view not good at all) there is no log message about declined login request - how should i implement a monitoring to track brute force attacks?
The only log message i get is the following
```
time="2024-08-01T07:59:08Z" level=debug msg="server: POST /api/v1/session (401) [1.612282ms]"
```

**Comment by lastzero** [comment]
Did you check if the issuer URI matches? What exact issuer does the nextcloud oidc api return through its ".well-known" endpoint? See our documentation for examples.

**Reply by sseide** [reply]
Hello,

Nextcloud is implementing the `https://nextcloud.my.domain/.well-known/openid-configuration` endpoint.
I already checked it and configured the exact same issuer id as reported there and as set with my other apps using Nextcloud as OIDC provider. Not working, no log messages...

And as i said - its missing the "Continue with OpenID" button on login page. I cannot if try to login with OIDC.

Btw what documentation do you mean? I only know these two pages:
* https://docs.photoprism.app/getting-started/advanced/openid-connect/#identity-providers
* https://docs.photoprism.app/developer-guide/api/oidc/

Are there more/other ones?

Thanks

**Reply by lastzero** [reply]
Does the issuer URL there also have a slash at the end as shown in your example? For google it's eg "https://accounts.google.com" without / at the end.

**Reply by sseide** [reply]
i tested booth versions (with and without). Example above was from second test with "/". 
Booth did not work. 

Nextcloud .well-known reports Issuer without slash, but (for unknown reasons) OpenProject for example required with "/" at the end.

**Comment by graciousgrey** [comment, marked as answer]
Hello :wave: ,

The problem might be that your configured PHOTOPRISM_SITE_URL uses http instead of https.

> there is no log message about declined login request 

You can view failed login attempts using the audit logins command:

https://docs.photoprism.app/user-guide/users/cli/#viewing-login-attempts

**Reply by sseide** [reply]
Yes - you are right. Changing the site url to "https://..." did the trick.

I set it to "http://..." i think because it is running behind an NGinx reverse proxy doing the TLS termination...
